tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,676 @@
+	'Neck', 
+	'Ring (left)', 
+	'Ring (right)', 
+	'Shield', 
+	'Shoulders', 
+	'Wrist'
+);
+
+CREATE TYPE DamageType AS ENUM (
+	'Bludgeoning', 
+	'Slashing', 
+	'Piercing', 
+	'Energy', 
+	'Acid', 
+	'Fire', 
+	'Electricity', 
+	'Cold', 
+	'Sonic', 
+	'Positive', 
+	'Negative', 
+	'Nonlethal'
+);
+
+CREATE TYPE Material AS ENUM (
+	'Nonspecific', 
+	'Abysium', 
+	'Adamantine', 
+	'Angelskin', 
+	'Aszite', 
+	'Magic Bridge Basalt', 
+	'Blight Quartz', 
+	'Bone', 
+	'Bronze', 
+	'Bulette', 
+	'Coral', 
+	'Clamshell', 
+	'Chitin', 
+	'Crypstone', 
+	'Blood Crystal',
+	'Darkleaf Cloth',
+	'Darkwood',
+	'Djezet',
+	'Dragonhide',
+	'Dragonskin',
+	'Druchite',
+	'Eel Hide',
+	'Elysian Bronze',
+	'Gold',
+	'Greenwood',
+	'Griffon Mane',
+	'Horacalcum',
+	'Inubrix',
+	'Cold Iron',
+	'Mindglass',
+	'Mithral',
+	'Noqual',
+	'Obsidian',
+	'Serpentstone',
+	'Siccatite',
+	'Alchemical Silver',
+	'Silversheen',
+	'Skymetal Alloy',
+	'Spiresteel',
+	'Fire-forged Steel',
+	'Frost-Forged Steel',
+	'Living Steel',
+	'Singing Steel',
+	'Stone',
+	'Sunsilver',
+	'Viridium',
+	'Voidglass',
+	'Whipwood',
+	'Wyroot'
+);
+
+--table declarations
+--user creds table - also used for hashing info
+CREATE TABLE UserInfo (
+	user_id     		UUID,
+
+        name        		TEXT,
+        is_admin    		BOOL,
+
+        pass_hash   		BYTEA,
+        salt        		BYTEA,
+
+        time_cost   		INT,
+        memory      		INT,
+        threads     		INT
+);
+
+--Character overview table
+CREATE TABLE Character (
+        user_id         	UUID,
+        char_id         	UUID,
+        race_id         	UUID,
+        deity_id        	UUID,
+
+        name            	TEXT,
+        age             	INT,
+        gender          	Gender,
+        alignment       	Alignment,
+        backstory       	TEXT,
+        height          	INT,
+        weight          	INT,
+        size            	Size,
+
+        strength        	INT,
+        dexterity       	INT,
+        constitution    	INT,
+        intelligence    	INT,
+        wisdom          	INT,
+        charisma        	INT,
+
+        max_hp          	INT,
+        damage          	INT,
+        nonlethal       	INT,
+
+        copper          	INT,
+        silver          	INT,
+        gold            	INT,
+        platinum        	INT
+);
+
+--Race tables
+CREATE TABLE Races (
+	race_id 		UUID,
+
+	name			TEXT,
+	move_speed		INT,
+	size			Size,
+	languages		TEXT,
+	race_type		RaceType
+);
+
+--Class tables
+CREATE TABLE Classes (
+	class_id		UUID,
+
+	name			TEXT,
+	hit_die			TEXT,
+	starting_wealth		TEXT,
+	bab_per_level		REAL,
+	skills_per_level	INT,
+	skills_attr		Attribute
+);
+
+CREATE TABLE Subclasses (
+	subclass_id		UUID,
+	class_id		UUID,
+
+	caster_type		CasterType,
+	casting_attr		Attribute
+);
+
+CREATE TABLE characterSubclasses (
+	char_id			UUID,
+	subclass_id		UUID,
+
+	levels_taken		INT,
+	hp_taken		INT,
+	skills_taken		INT
+);
+
+--Feats tables
+CREATE TABLE Feats (
+	feat_id			UUID,
+	req_id			UUID,
```

Comment:
What if a feat has multiple requirements?

Reply:
fixed; requirements have been changed to reference the feat_id primary key, allowing a feat to have multiple requirements.